src/SensioLabs/Melody/Handler/FileHandler.php

```diff
@@ -36,6 +36,11 @@ public function createResource($filename)
             sprintf('file://%s', realpath($filename))
         );
 
-        return new LocalResource($filename, file_get_contents($filename), $metadata);
+        $content = file_get_contents($filename);
+        if (0 !== strpos('#!', $content)) {
+            return new LocalResource($filename, $content, $metadata);
+        }
+
+        return new LocalResource($filename, preg_replace('/^#![^\n]+\n/u', '', $content), $metadata, true);
```

Comment:
given you already check that the string starts with `#!`, what's about spliting on `\n` and keeping the 2ns part?

```php
explode("\n", $content, 2)[1] ?? ''
# or to avoid the `??` that preserve a failure on a script thatn contains only a shebang
explode("\n", $content."\n", 2)[1]
```